packages/shared/lib/core/wallet/classes/activity.class.ts

```diff
@@ -96,106 +47,69 @@ export class Activity implements IActivity {
-            const isInternal = isSubjectInternal(subject)
-            this.isInternal = isInternal
-
-            this.sender = sender
-            this.recipient = recipient
-            this.subject = subject
-
-            this.direction = isIncoming || isSelfTransaction ? ActivityDirection.In : ActivityDirection.Out
+        this.id = transactionId
 
-            this.isAsync = isOutputAsync(output)
-            this.isClaimed = false
-            this.asyncStatus = this.isAsync ? this.getAsyncStatus(new Date()) : null
-
-            const nativeToken = getNativeTokenFromOutput(output)
-
-            this.assetId = nativeToken?.id ?? String(COIN_TYPE[get(activeProfile).networkProtocol])
-            const asset = getPersistedAsset(this.assetId)
-            this.isAssetHidden = !asset || asset.hidden
-            this.isRejected = isActivityHiddenForAccountId(account.id, transactionId)
-
-            const { storageDeposit, giftedStorageDeposit } = getStorageDepositFromOutput(output)
-            this.storageDeposit = storageDeposit
-            this.giftedStorageDeposit = giftedStorageDeposit
-            this.rawAmount = nativeToken
-                ? Number(nativeToken?.amount)
-                : getAmountFromOutput(output) - this.storageDeposit
-
-            this.metadata = getMetadataFromOutput(output)
-            this.tag = getTagFromOutput(output)
-            this.expirationDate = getExpirationDateFromOutput(output)
+        if (type === ActivityType.Transaction) {
+            this.data = getTransactionActivityData(processedTransaction, account)
         } else if (type === ActivityType.Foundry) {
-            const { output, outputIndex } = getFoundryOutputFromTransaction(outputs)
-            this.id = outputIdFromTransactionData(transactionId, outputIndex)
-            this.outputId = outputIdFromTransactionData(transactionId, outputIndex) // not required for foundry activities
-
-            const nativeToken = getNativeTokenFromOutput(output)
-
-            this.assetId = nativeToken?.id ?? String(COIN_TYPE[get(activeProfile).networkProtocol])
-            const asset = getPersistedAsset(this.assetId)
-            this.isAssetHidden = !asset || asset.hidden
-
-            const { storageDeposit, giftedStorageDeposit } = getStorageDepositFromOutput(output) // probably we need to sum up all storage deposits
-            this.storageDeposit = storageDeposit
-            this.giftedStorageDeposit = giftedStorageDeposit
-            this.rawAmount = nativeToken
-                ? Number(nativeToken?.amount)
-                : getAmountFromOutput(output) - this.storageDeposit
+            this.data = getFoundryActivityData(processedTransaction)
         }
     }
 
     updateFromPartialActivity(partialActivity: Partial<IActivity>): void {
         Object.assign(this, partialActivity)
     }
 
+    updateDataFromPartialActivity(partialData: Partial<TransactionActivityData | FoundryActivityData>): void {
```

Comment:
for better dev-ex, i'd like to have the `type` field required in this parameter, do you have any suggestions how to do that?